.github/workflows/windows-rubygems.yml

```diff
@@ -29,22 +29,22 @@ jobs:
           - { name: "2.7", value: 2.7.6 }
           - { name: "3.0", value: 3.0.4 }
           - { name: "3.1", value: 3.1.2 }
-          - { name: mswin, value: mswin }
+          - { name: head, value: head }
```

Comment:
Happy to change, but given that mswin is a head build, including both might clarify the issue when failures occur?

Reply:
Oh, really? I didn't know that. Let's move mswin to the daily CI then, since that's where we keep moving targets.

Reply:
I'll look at the daily workflows.  JFYI, setup-ruby has four Windows head builds - head, mingw, mswin, & ucrt.

Reply:
A quick look, it looks like there isn't any WIndows or macOS runs in the 'daily' workflows.  Not.sure.

Maybe take the windows-rubygems.yml as it is here and divide it, creating a daily-windows-rubygems.yml workflow?

Reply:
Also, re head builds, Ubuntu-22.04 image is available, it's using OpenSSL 3.  So, Ruby 3.1 and head are available on it.

Windows mswin is also using OpenSSL 3.

Reply:
We should add an "os" matrix that includes windows to the `daily-rubygems.yml` workflow I think.

Reply:
Okay, I think that should work.

Maybe:

ubuntu-22.04 head
ubuntu-20.04 head, truffleruby-head

macos-12 head
not sure about macos-11

windows-2022 head, mswin

Thoughts?

Regardless, it will be a bit later.

Reply:
Two items:

> so far Windows daily CI has never caught any issues


I must be missing something, but I see two cron workflows, `daily-bundler.yml` & `daily-rubygems.yml`.  They only run on ubuntu-20.04?

2nd, all the Actions images have Rust/Cargo installed.  It looks like they update it as they release new images.  Right now, I think all images are on 1.62.0, but 1.62.1 was released after the images.  IOW, I think the next images updates will have 1.62.1.

So, would you prefer to continue to update Rust/Cargo, or should we just use what the images have installed?

Reply:
> I must be missing something, but I see two cron workflows, daily-bundler.yml & daily-rubygems.yml. They only run on ubuntu-20.04?

You're right! I still the current matrix is enough for us if you don't mind.

> So, would you prefer to continue to update Rust/Cargo, or should we just use what the images have installed?

I'm not sure, but I prefer less moving targets so I'd go with the default version. Not sure what @ianks thinks.

Reply:
If you look at the workflow run for this PR, you'll see that the Ruby MSYS2 ucrt head build installs:
```
info: installing component 'rust-mingw'
```
but the mswin build does not.  That was the reason for the PR, to use the target `x86_64-pc-windows-gnu` for MSYS2 MINGW64 or UCRT64 builds, and `x86_64-pc-windows-msvc` for the mswin build, which is built with Visual Studio 2022.

<details>
<summary>Some of the actions-rs/toolchain logs for both jobs are listed below:</summary>

The [head job](https://github.com/rubygems/rubygems/runs/7432084294?check_suite_focus=true#step:4:45) shows:
```
info: latest update on 2022-07-19, rust version 1.62.1 (e092d0b6b 2022-07-16)
info: downloading component 'cargo'
info: downloading component 'rust-mingw'
info: downloading component 'rust-std'
info: downloading component 'rustc'
info: installing component 'cargo'
info: installing component 'rust-mingw'
info: installing component 'rust-std'
info: installing component 'rustc'
  stable-x86_64-pc-windows-gnu installed - rustc 1.62.1 (e092d0b6b 2022-07-16)
```

The [mswin job](https://github.com/rubygems/rubygems/runs/7432084464?check_suite_focus=true#step:4:78) shows:
```
info: latest update on 2022-07-19, rust version 1.62.1 (e092d0b6b 2022-07-16)
info: downloading component 'clippy'
info: downloading component 'rustfmt'
info: downloading component 'rust-std' for 'i686-pc-windows-msvc'
info: downloading component 'cargo'
info: downloading component 'rust-std'
info: downloading component 'rustc'
info: removing previous version of component 'clippy'
info: removing previous version of component 'rustfmt'
info: removing previous version of component 'rust-std' for 'i686-pc-windows-msvc'
info: removing previous version of component 'cargo'
info: removing previous version of component 'rust-std'
info: removing previous version of component 'rustc'
info: installing component 'clippy'
info: installing component 'rustfmt'
info: installing component 'rust-std' for 'i686-pc-windows-msvc'
info: installing component 'cargo'
info: installing component 'rust-std'
info: installing component 'rustc'
  stable-x86_64-pc-windows-msvc updated - rustc 1.62.1 (e092d0b6b 2022-07-16) (from rustc 1.62.0 (a[83](https://github.com/rubygems/rubygems/runs/7432084464?check_suite_focus=true#step:4:84)14ef7d 2022-06-27))
```
</details>

Reply:
Yes, I understood that and it seems fine to me. But I don't understand why we need more moving targets.

Can we change the target for the `mswin` build as you mentioned but not introduce `head`? Have we introduced any bugs in Windows recently that would've been avoided if we had tested against `head`? Reversely, the `mswin` has not detected any issues either that I recall, so I'm not sure why `head` would. I feel we're just growing our matrix for no good reason.

In any case, if you feel very strongly, as long as these are moved to a daily CI, I can live with it.

Reply:
> if you feel very strongly

First of all, I try to keep written communication short and to the point.  That can be mistaken for feeling very strongly.

Secondly, a long time ago, CI Ruby master builds were created without regard for whether they passed tests.  That's not the case anymore, so, no, I don't have strong feelings.

> we're just growing our matrix for no good reason.

Which uses resources that are free and more electricity.  Again, mswin is fine.

Maybe wait to see if @ianks has thoughts, then move mswin to the daily.  Maybe it isn't about the Ruby builds, but about testing both targets (`x86_64-pc-windows-gnu` and `x86_64-pc-windows-msvc`)...

Reply:
Ok, let's wait for @ianks input, yeah!

Reply:
So I have not yet worked on making things stable for `x86_64-pc-windows-msvc`, yet. Mostly because I was not sure how to get one installed in GitHub actions. If there is an easy way to do so now, I'd be more than happy to see if we can get Cargo to play nicely with it. 

~It's probably pretty close already.~ Oh wow! It looks like it's already passing. I'd be happy to do this and support msvc once we add it to the [`rb-sys` CI matrix](https://github.com/oxidize-rb/rb-sys/blob/main/.github/workflows/ci.yml#L50)

Reply:
@ianks

> once we add it to the [rb-sys CI matrix](https://github.com/oxidize-rb/rb-sys/blob/main/.github/workflows/ci.yml#L50)

Thanks for the reply.  I'm rather Rust challenged, but I thought I'd try.  Not passing.  See CI job in my fork:

https://github.com/MSP-Greg/rb-sys/runs/7442560121?check_suite_focus=true#step:4:118

Reply:
Ok, so let's keep this PR opened until rb-sys tests with `x86_64-pc-windows-msv` and I'm happy to accept separate PRs to:

* Move `mswin` to the daily CI.
* Stop updating rust and let the default be used (if @ianks thinks it's fine).

Reply:
https://github.com/oxidize-rb/rb-sys/issues/71

Reply:
mswin moved to daily CI at #5760.